src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.runconfig.test
+
+import com.google.gson.*
+import com.intellij.execution.testframework.TestConsoleProperties
+import com.intellij.execution.testframework.sm.ServiceMessageBuilder
+import com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter
+import com.intellij.openapi.util.Key
+import jetbrains.buildServer.messages.serviceMessages.*
+
+data class LibtestSuiteMessage(
+    val type: String,
+    val event: String
+) {
+    companion object {
+        fun fromJson(json: JsonObject): LibtestSuiteMessage? {
+            if (json.getAsJsonPrimitive("type")?.asString != "suite") {
+                return null
+            }
+
+            return Gson().fromJson(json, LibtestSuiteMessage::class.java)
+                ?: error("Failed to parse LibtestSuiteMessage from $json")
```

Comment:
Maybe it's better to log cases when we can't convert json into object instead of throwing exception?
Also, as I can see, `Gson#fromJson` throws `JsonSyntaxException` instead of returning null, so looks like `error` function won't be really called at all